torchvision/prototype/transforms/_misc.py

```diff
@@ -32,6 +35,68 @@ def extra_repr(self) -> str:
         return ", ".join(extras)
 
 
+class LinearTransformation(Transform):
+
+    _transformed_types: Tuple[Type, ...] = (torch.Tensor, features.Image, PIL.Image.Image)
+
+    def __init__(self, transformation_matrix: torch.Tensor, mean_vector: torch.Tensor):
+        super().__init__()
+        if transformation_matrix.size(0) != transformation_matrix.size(1):
+            raise ValueError(
+                "transformation_matrix should be square. Got "
+                f"{tuple(transformation_matrix.size())} rectangular matrix."
+            )
+
+        if mean_vector.size(0) != transformation_matrix.size(0):
+            raise ValueError(
+                f"mean_vector should have the same length {mean_vector.size(0)}"
+                f" as any one of the dimensions of the transformation_matrix [{tuple(transformation_matrix.size())}]"
+            )
+
+        if transformation_matrix.device != mean_vector.device:
+            raise ValueError(
+                f"Input tensors should be on the same device. Got {transformation_matrix.device} and {mean_vector.device}"
+            )
+
+        self.transformation_matrix = transformation_matrix
+        self.mean_vector = mean_vector
+
+    def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
+
+        pil_mode = None
+        if isinstance(inpt, PIL.Image.Image):
+            pil_mode = inpt.mode
+            inpt = F.to_image_tensor(inpt).contiguous()
+
+        shape = inpt.shape
+        n = shape[-3] * shape[-2] * shape[-1]
+        if n != self.transformation_matrix.shape[0]:
+            raise ValueError(
+                "Input tensor and transformation matrix have incompatible shape."
+                + f"[{shape[-3]} x {shape[-2]} x {shape[-1]}] != "
+                + f"{self.transformation_matrix.shape[0]}"
+            )
+
+        if inpt.device.type != self.mean_vector.device.type:
+            raise ValueError(
+                "Input tensor should be on the same device as transformation matrix and mean vector. "
+                f"Got {inpt.device} vs {self.mean_vector.device}"
+            )
+
+        flat_tensor = inpt.view(-1, n) - self.mean_vector
+        transformed_tensor = torch.mm(flat_tensor, self.transformation_matrix)
+        output = transformed_tensor.view(shape)
+
+        if pil_mode is not None:
+            # cast output to input dtype:
+            output = output.to(inpt.dtype)
+            output = F.to_image_pil(output, mode=pil_mode)
+        elif isinstance(inpt, features.Image):
+            output = features.Image.new_like(inpt, output)
```

Comment:
Given that the linear transform is going to move the values away from the standard scale, making it impossible for us to tell what is the maximum value (needed for many transforms), I think we shouldn't wrap things back as images (PIL or `_feature.Image`) but return things as tensors. Thoughts?

Reply:
Actually, I think I should drop PIL Image support as stable LinearTransformation does not support it: https://github.com/pytorch/vision/blob/a409c3fef429076168747d375614e10e72cf035f/torchvision/transforms/transforms.py#L1072-L1079